### ISSUE TITLE
Chimera removal after denoising

### ISSUE THREAD

**Issue by clairexu010** [benjjneb/dada2#1154]
Hi Benjamin,

On the DADA2 tutorial website, I found a FAQ - “Why are so many of my reads being removed as chimeras?”
 
In the answer, you said:
 
**“It is common for chimeric sequences to be a majority (even large majority) of the total number of unique sequence variants inferred by the dada(...) algorithm, but they should not be a majority of the sequencing reads (<25%).”**
 
May I ask **why is it reasonable majority of ASVs are inferred as chimeras when using DADA2 for denoising**? **Why is 25% selected as the threshold?**

The reason I ask this is, in my dataset, over 97% ASVs are inferred as chimeras and removed, which accounts for 19% of merged reads. This doesn't hit the threshold. But I'm still wondering the reason for high proportion of chimeras and why it is reasonable.

BTW, following your suggestion in the tutorial, high amount of chimera might be due to the inappropriate removal of primer. So I double-checked the reads on 5'-end for both Read1 and Read2. **There is no primer sequence.**

Would be greatly appreciated to have your comments.

Thank you~

**Comment by benjjneb** [comment]
> May I ask why is it reasonable majority of ASVs are inferred as chimeras when using DADA2 for denoising?

I think I would rephrase this question as why is it reasonable that a majority of ASVs are chimeras. The answer is that (especially with multi-V-region amplicons) that chimeras can produce a "combinatorial" number of new sequences -- that is they can paste together many different combinations of "right sequences" and "left sequences" such that their total number in terms of unique ASVs can exceed that of the real sequences in the samples.

> Why is 25% selected as the threshold?

First, the process hinted at above can produce a large number of low abundance ASVs, that is there are many chimeras that exists only in a few reads in the dataset. Thus, identifying and removing them reduces the total ASVs by a lot, but not the totla read by a lot. The "25%" threshold is an imperfect, but educated, rule of thumb for determining whether there are too many chimeric reads in your data. Repeatedly it has been shown that PCR chimeras can approach 25% (perhaps 30%?) of the data, albeit in poorly designed/optimized PCRs. However, above that level is rarely in our experience seen in practice.

Thus 25% as a simple empirical threshold.

> The reason I ask this is, in my dataset, over 97% ASVs are inferred as chimeras and removed, which accounts for 19% of merged reads. This doesn't hit the threshold. But I'm still wondering the reason for high proportion of chimeras and why it is reasonable.

At the merged reads level, that is not an extreme value. But 97% of ASVs chimeric is high in practice, but not necessarily wrong. Can you describe more about the amplicon setup you have? What primers, what library preparation etc.

**Comment by clairexu010** [comment]
Hi Ben, 

Thank you so much for your prompt and detailed explanation about chimera. Please find the sequencing and library preparation information of my dataset below.

- Samples: ~1600 human stool sample at multiple timepoints

- Sequencing: 16S V4 region, 250 paired-end, sequenced on 4 lanes, 2 runs on Hiseq 2500

- Library preparation: 
![image](https://user-images.githubusercontent.com/72391137/95408860-04a83480-0953-11eb-9b23-617d62186af5.png)

    In our scenario, the forward and reverse primers were used as sequencing primers. And actually we had another dataset, which didn't use primer as the sequencing primer. We still obtained high proportion of chimeras (83% of merged ASVs, 20% of merge reads). 

- **Fastqc**

![image](https://user-images.githubusercontent.com/72391137/95409173-aaf43a00-0953-11eb-80ec-7f53b6157683.png)

- **QC**
truncQ = 2, minLen = 150, maxEE = 1, truncLen = c(200,150) (Read1 & 2)

    We also truncate Read1 at the 3'-end because Read1 and 2 are nearly fully overlapped for V4 region. As the length for V4 region is variable, for certain bacteria, it might be <250bp. To improve merging and merge correctly, we truncate at 3'-end for Read1. And there would be still around 50bp for overlapping.

- **Unique reads/ASVs**

![image](https://user-images.githubusercontent.com/72391137/95409864-33270f00-0955-11eb-9712-43c7179f22f8.png)

- **Reads kept for each step**
![image](https://user-images.githubusercontent.com/72391137/95409915-49cd6600-0955-11eb-8f28-32d5e0528f84.png)

If I miss anything, please let me know. Hope to have your comments.

**Comment by clairexu010** [comment]
To add on, actually I also try to process the data using USEARCH on a randomly selected 100 samples. Previously I was always using USERACH for 16S data processing. Finally, 1219 OTUs (97% similarity) are obtained. Among them, 99 are inferred as chimeras. Very strange here. Seems using traditional OTU clustering method, less chimeras would be inferred compared using DADA2. Is it because the algorithm difference between DADA2 and UPARSE?

**Comment by benjjneb** [comment]
For V4, you're percentage merging (read-wise) is also not as high as usual.
Is there any other potential technical issue with your sequnecing?

One possibility is that your primers are of variable length, or that heterogeneity spacers were used in the library preparation?
Also, why are there two amplicon described in your library figure, the regular and prime' versions. Do they differ in some way?

It might also be useful to post the first few sequences from an example R1 read file.

**Comment by clairexu010** [comment]
> For V4, you're percentage merging (read-wise) is also not as high as usual.
    Yes, we lose around 10% of reads after merging.
> Is there any other potential technical issue with your sequnecing?
> One possibility is that your primers are of variable length, or that heterogeneity spacers were used in the library preparation?
    The sequencing was done by a company. I've double-checked with them, there is no heterogeneity spacer during library preparation. Here is the detailed library preparation. Not sure whether it would help. 
   There are tow modifications compared with a regular library preparation:
    - They link the adaptor, index and primer into one, and there is only one PCR preparation step.
    - They use the forward and reverse primer as the sequencing primer.

This is the length distribution of all samples. They are all 250bp.
![image](https://user-images.githubusercontent.com/72391137/95699141-477d4b80-0c76-11eb-9c5d-13d0f7c1bc2c.png)
![image](https://user-images.githubusercontent.com/72391137/95699159-519f4a00-0c76-11eb-8f14-f885796d7c76.png)

This is the sequence content. The first 8bp are exactly the conserved region of V4.
![image](https://user-images.githubusercontent.com/72391137/95699032-0422dd00-0c76-11eb-87b3-1c30640d5a4f.png)
![image](https://user-images.githubusercontent.com/72391137/95699095-2e749a80-0c76-11eb-8391-42562dbe9179.png)

> Also, why are there two amplicon described in your library figure, the regular and prime' versions. Do they differ in some way?
   The figure shows forward and reverse read, rather than two amplicons. Hope I did not misunderstand your point.

> It might also be useful to post the first few sequences from an example R1 read file.
    Here is the screenshot of one sample's R1 read file
![image](https://user-images.githubusercontent.com/72391137/95697416-3da51980-0c71-11eb-95a6-b960c8a081f5.png)

**Comment by clairexu010** [comment]
@benjjneb 
Another strange thing is the number of merged ASVs is higher than number of either forward or reverse ASVs. Is this possible?
![image](https://user-images.githubusercontent.com/72391137/95724727-49accd80-0ca9-11eb-8112-bd1a1d958a95.png)

**Comment by benjjneb** [comment]
I'm not sure anything is wrong here. Your percent of reads merging and percent of reads removed as chimeras are a bit higher than I usually see with V4 data, but that could be a real thing due to a higher-chimera PCR protocol.

> Another strange thing is the number of merged ASVs is higher than number of either forward or reverse ASVs. Is this possible?

Yes. For example, if variation between two ASVs occurs only in the part of the amplicon covered only by the forward read, or only by the reverse read, it can result in multiple merged ASVs from a single reverse (or forward) ASV.

**Comment by clairexu010** [comment]
Thank you so much for your detailed comments @benjjneb 

> I'm not sure anything is wrong here. Your percent of reads merging and percent of reads removed as chimeras are a bit higher than I usually see with V4 data, but that could be a real thing due to a higher-chimera PCR protocol.
   
   I also suspect the PCR protocol might be the reason for higher chimera. 

- However, in my opinion, as long as the chimera doesn't affect the microbial composition (sorry, I didn't find any reference discussing about whether chimera would affect the microbial community or not. If there has, please correct me) and the sequencing is deep enough to cover most of the diversity (Chao1 and rarefaction reached a plateau), we can still go on with the analysis after removing chimera.

- Another is, the average sequencing depth is quite high in our study. The samples have an average of 86550 reads. We have around 18% of the samples are from infants at early time point. The DNA yield and diversity are very low for these samples. So we try to sequence as deeply as we can to cover the rare species. I suspect the signal of chimera would also be enlarged at such a sequencing depth. 

   That's my opinion. Would be great to have your comment.

**Comment by benjjneb** [comment]
> would either forward ASV index 1 or reverse ASV index 1 also have abundance of 24527?

Agreed.

> I suspect the signal of chimera would also be enlarged at such a sequencing depth.

Chimeras tend to occur at low frequencies, so this is probably true to some degree, although I've not tried to quantify the magnitude of this effect.

**Comment by clairexu010** [comment]
Sorry Ben, I might miss your point. Possibly you were answering to someone else below.
> > would either forward ASV index 1 or reverse ASV index 1 also have abundance of 24527?
> 
> Agreed.
>

**Comment by benjjneb** [comment]
Oops! yes, copy/paste error there. Should have read:

> we can still go on with the analysis after removing chimera.

Agreed.
